src/guide/composition-api-provide-inject.md

```diff
@@ -1,120 +1,261 @@
+      longitude: 90,
+      latitude: 135
+    })
+
+    provide('location', location)
+    provide('geolocation', geolocation)
 
     return {
-      book,
-      year
+      location
+    }
+  },
+  methods: {
+    updateLocation() {
+      this.location = 'South Pole'
     }
   }
 }
+</script>
 ```
 
-## Injection Reactivity
+However, there are times where we need to update the data inside of the component where the data is injected. In this scenario, we recommend providing a method that is responsible for mutating the reactive property.
 
-To retain reactivity between provided and injected values, we can use a [ref](reactivity-fundamentals.html#creating-standalone-reactive-values-as-refs) or [reactive](reactivity-fundamentals.html#declaring-reactive-state) when providing a value:
+```vue{21-23,27}
+<!-- src/components/MyMap.vue -->
+<template>
+  <MyMarker />
+</template>
 
-```js
-import { ref, reactive } from 'vue'
-
-// in provider
-setup() {
-  const book = reactive({
-    title: 'Vue 3 Guide',
-    author: 'Vue Team'
-  })
-  const year = ref('2020')
-
-  provide('book', book)
-  provide('year', year)
-}
+<script>
+import { provide, reactive, ref } from 'vue'
+import MyMarker from './MyMarker.vue
 
-// in consumer
-setup() {
-  const book = inject('book')
-  const year = inject('year')
+export default {
+  components: {
+    MyMarker
+  },
+  setup() {
+    const location = ref('North Pole')
+    const geolocation = reactive({
+      longitude: 90,
+      latitude: 135
+    })
+
+    const updateLocation = () => {
+      location.value = 'South Pole'
+    }
 
-  return { book, year }
+    provide('location', location)
+    provide('geolocation', geolocation)
+    provide('updateLocation', updateLocation)
+  }
 }
+</script>
 ```
 
-Now, when either `book` or `year` are changed on the _provider_ component, we can observe them changing on the component where they are injected.
-
-::: warning
-We don't recommend mutating a reactive property where it's injected as it's breaking Vue one-direction data flow. Instead, try to either mutate values where they are _provided_ or provide a method to mutate them
-
-```js
-import { ref, reactive } from 'vue'
+```vue{9,14}
+<!-- src/components/MyMarker.vue -->
+<script>
+import { inject } from 'vue'
 
-// in provider
-setup() {
-  const book = reactive({
-    title: 'Vue 3 Guide',
-    author: 'Vue Team'
-  })
+export default {
+  setup() {
+    const userLocation = inject('location', 'The Universe')
+    const userGeolocation = inject('geolocation')
+    const updateUserLocation = inject('updateUserLocation')
 
-  function changeBookName() {
-    book.title = 'Vue 3 Advanced Guide'
+    return {
+      userLocation,
+      userGeolocation,
+      updateUserLocation
+    }
   }
-
-  provide('book', book)
-  provide('changeBookName', changeBookName)
 }
+</script>
+```
 
-// in consumer
-setup() {
-  const book = inject('book')
-  const changeBookName = inject('changeBookName')
+:::tip
```

Comment:
suggestion: To remove the `:::tip` and blend this content into the current flow. WDYT?

Reply:
I was debating this too. With your recommendation, let me blend it in better